cms/djangoapps/contentstore/views/tests/test_credit_eligibility.py

```diff
@@ -0,0 +1,69 @@
+"""
+Unit tests for credit eligibility methods.
+"""
+from mock import patch
+
+from contentstore.tests.utils import CourseTestCase
+from student.tests.factories import UserFactory
+from util.testing import UrlResetMixin
+from xmodule.modulestore.tests.factories import CourseFactory
+
+
+class CreditEligibilityViewTest(UrlResetMixin, CourseTestCase):
+    """
+    Base class to test the credit eligibility view.
+    """
+    def setUp(self, **kwargs):
+        enable_credit_eligibility = kwargs.get('enable_credit_eligibility', False)
+        with patch.dict('django.conf.settings.FEATURES', {'ENABLE_CREDIT_ELIGIBILITY': enable_credit_eligibility}):
+            super(CreditEligibilityViewTest, self).setUp('cms.urls')
+
+        self.user = UserFactory(is_staff=True)  # pylint: disable=no-member
+        self.client.login(username=self.user.username, password='test')
+        self.course = CourseFactory.create(org='edX', number='dummy', display_name='Credit Course')
+        self.credit_eligibility_url = (
+            '/view_credit_eligibility'
+            '/{course_key}'
+        ).format(course_key=unicode(self.course.id))
```

Comment:
@rlucioni we cannot use reverse here since `NoReverseMatch` exception will be raised when `ENABLE_CREDIT_ELIGIBILITY` feature flag is not disabled as `view_credit_eligibility` url is conditionally added.